projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -1044,6 +1041,13 @@ function get_paywall_content( $post_access_level, $email_confirmation_pending =
  * @return string
  */
 function get_paywall_blocks( $newsletter_access_level ) {
+	$custom_paywall = apply_filters( 'jetpack_custom_paywall_blocks', false );
+	if ( ! empty( $custom_paywall ) ) {
+		return $custom_paywall;
```

Comment:
Would it make sense to pass same conditionals as below, or even wrap the final block-content output into this filter?

Reply:
Thinking further, suppose they have access to same conditional checks (logged/paying/etc) when they hook into the filter anyway. :-)

How you implemented this might actually be cleanest at the end for now.